packages/liveblocks-node/src/client.ts

```diff
@@ -48,6 +58,48 @@ type ThreadParticipants = {
   participantIds: string[];
 };
 
+export type RoomPermission =
+  | ["room:write"]
+  | ["room:read", "room:presence:write"];
+export type RoomAccesses = Record<string, RoomPermission>;
+export type RoomMetadata = Record<string, string | string[]>;
+
+export type RoomInfo = {
+  type: "room";
+  id: string;
+  lastConnectionAt?: string;
+  createdAt?: string;
+  metadata: RoomMetadata;
+  groupsAccesses: RoomAccesses;
+  usersAccesses: RoomAccesses;
+  defaultAccesses: RoomPermission;
+  schema?: string;
+};
+
+export type RoomUser<Info> = {
+  type: "user";
+  id: string | null;
+  connectionId: number;
+  info: Info;
+};
+
+export type YJson =
+  | {
+      [x: string]: unknown;
+    }
+  | string
+  | undefined
+  | unknown[];
+
+export type Schema = {
+  id: string;
+  name: string;
+  version: number;
+  createdAt: string;
+  updatedAt: string;
```

Comment:
I think it will be nicer if we'd convert these back into Date instances for our users.

```suggestion
  createdAt: Date;
  updatedAt: Date;
```

This means we can no longer directly return `(await res.json()) as Schema`, but we'd have to convert these fields explicitly. Doing so at least prevents us from having to "fix" this type/return value later in change that will be a breaking change at that point.

If we have more exported types here with dates in them, I'd suggest to do the same for all.

Reply:
> The Comments functions in `@liveblocks/node` return [`ThreadData`](https://github.com/liveblocks/liveblocks/blob/a7aef1d0a0cf8b5bada932805ef52d00cccc77ef/packages/liveblocks-core/src/comments/types/ThreadData.ts#L11) which uses a `string`. This same type is also returned on the front end by `useThreads`.

Is this something can we can still change for the comments use case?

Keeping and communicating the semantics of fields in APIs is really important and sometimes the name of those fields just aren't enough to really know what values are legal. While technically the fields _must_ travel over the wire as strings (thanks JSON spec), our `@liveblocks/node` library is in the unique position to translate them back to a semantic value before returning them to the user.

This way, the type that users get exposed to can simply keep using Date reliably, and we don't have to expose our users to the implementation details of our serialization format directly.

Reply:
I like the idea of converting the strings to Date before returning. Like @nvie said, this allows the library to manage the complexity of serialization and deserialization while providing a more intuitive and useful interface to the users. How do you suggest we go about this if we do? `getThreads`, `getThread`, `getComment` currently exposes `createdAt` property as a string (and `useThreads` from our react package does so too). Having them also return a Date instead would mean breaking change, I believe. Do we want to do it for all methods (including `getRoom`, `getRooms`) in this PR or a different PR?

Reply:
I think a breaking change is legitimized to fix an existing issue with this format (and since comments is still in beta, I think it's fine), as long as we document it, and tackle all the instances at once. I think most users won't run into problems with it.

Reply:
What I have done for now is for non-comments related methods (like `getRoom`, `createRoom`, `getRooms`, `getSchema`, `createSchema`), the properties like `createdAt`, etc that are returned as ISO strings from server are converted to `Date` objects before being returned. 

The same isn't true for comments related methods yet because I noticed a lot of dependency of comment types in other packages and think that change warrants a separate PR of its own.

https://github.com/liveblocks/liveblocks/blob/a7243a992d377e01223dcab1db49a6862d450327/packages/liveblocks-node/src/client.ts#L400-L422

Reply:
I agree with making a separate PR for that, and regarding doing the same in the front-end packages, given that the `Date` constructor and our `<Timestamp />` primitive both accept `string` and `Date` (and `number`), the breaking change transition could be very minimal.